### PR TITLE
add connection id

### PR DESCRIPTION
**Pull request by uri-p**

### CODE REVIEW

File: src/lumigo_tracer/spans_container.py [hunk 1]
```diff
@@ -73,6 +74,7 @@ def __init__(
         self.transaction_id = transaction_id
         self.max_finish_time = max_finish_time
         self.base_msg = {
+            "lambda_container_id": SpansContainer.lambda_container_id,
```
Comment (saartochner) [inline src/lumigo_tracer/spans_container.py:77]: do you want it in the HTTP spans as well?
If no, you should put it under seelf.function_span

File: src/lumigo_tracer/wrappers/http/http_data_classes.py [hunk 2]
```diff
@@ -8,13 +8,15 @@ class HttpRequest:
...
     body: bytes
+    instance_id: bytes
 
     def __init__(self, **kwargs):
         self.host = kwargs["host"]
         self.method = kwargs["method"]
         self.uri = kwargs["uri"]
         self.headers = {k.lower(): v for k, v in (kwargs.get("headers") or {}).items()}
         self.body = kwargs.get("body")
+        self.instance_id = kwargs.get("instance_id")
```
Comment (saartochner) [inline src/lumigo_tracer/wrappers/http/http_data_classes.py:19]: self.instance_id maybe should be `Optional[bytes]`?

File: src/lumigo_tracer/wrappers/http/sync_http_wrappers.py [hunk 3]
```diff
@@ -175,10 +175,19 @@ def _http_send_wrapper(func, instance, args, kwargs):
...
+                    uri=uri,
+                    headers=headers,
+                    body=body,
+                    instance_id=id(instance),
+                )
             )
         else:
-            add_unparsed_request(HttpRequest(host=host, method=method, uri=uri, body=data))
+            add_unparsed_request(
+                HttpRequest(host=host, method=method, uri=uri, body=data, instance_id=id(instance))
```
Comment (saartochner) [inline src/lumigo_tracer/wrappers/http/sync_http_wrappers.py:189]: what is an `id` of an `instance` here?
This is an instance of an `HTTPConnection` which doesn't tell you much about the request/session/machine 🤔 
What are you trying to achieve?

**Review by saartochner — CHANGES_REQUESTED** [review]
GJ!

File: src/lumigo_tracer/spans_container.py (hunk 1 above)
Comment (uri-p) [reply src/lumigo_tracer/spans_container.py:77]: Yes for all connections

File: src/lumigo_tracer/wrappers/http/sync_http_wrappers.py (hunk 3 above)
Comment (uri-p) [reply src/lumigo_tracer/wrappers/http/sync_http_wrappers.py:189]: The goal is to validate the reuse of connections.
For example:
Multiple connections to the same AWS service like kinesis lambda etc.

The data is based on HTTP/S connection id + host

File: src/lumigo_tracer/wrappers/http/http_data_classes.py (hunk 2 above)
Comment (uri-p) [reply src/lumigo_tracer/wrappers/http/http_data_classes.py:19]: Is it the same as 'body'?
I don't see any optional

Comment (saartochner) [reply src/lumigo_tracer/wrappers/http/http_data_classes.py:19]: You right, change body too :)

**Review by saartochner — APPROVED** [review]

**Pull request merged**